### ISSUE TITLE
SQL Collison Possibilities

### ISSUE THREAD

**Issue by KOOKIIEStudios** [TEAM-SPIRIT-Productions/Lazuli#1]
**Describe the bug**
Currently the setter methods for `username` (in `accounts`) and `name` (in `characters`) allow changes without collision checks.

**To Reproduce**
1. Change the `Account(account_info, database_config).username` or `Character(char_stats, database_config).name` to an existing name
2. See DB throwing errors for account name (key) clashes on console, but not for user. Notice that DB allows collisions for character names (non-key).

**Expected behavior**
Consider making such collisions raise an exception that can be carried forward using try-catch

**Screenshots**
Notice name collision in character IGNs:
![image](https://user-images.githubusercontent.com/25145447/103482269-e602ee00-4e1a-11eb-8666-547fd4be7a9a.png)

**Comment by KOOKIIEStudios** [comment]
## Consider the following demo:
```py
class Character:  # To simulate Lazuli's character class
	def __init__(self, name):
		self._name = name
	
	@property
	def name(self):
		return self._name
		
	@name.setter
	def name(self, new_name):
		if new_name == self._name:  # checks for setting same name as before here, but it
		# should check for DB matches in the actual engine instead
			raise ValueError("That name is already taken!")  # Message to be passed along on failure
		else:
			self._name = new_name
	
print("Creating character 'soul'")
soul = Character("soul")
print(f"  Name: {soul.name}")
print("Setting name to clash")
try:  # Simulate dependents using the setter following command issue from Discord bot
	soul.name = "soul"
except Exception as e:
	print("Error encountered")
	if type(e) is ValueError:
		# `print` used here, to simulate Discord message sending (by the bot)
		print(f"  Carrying over error message: {e.args[0]}")  # Message to be passed along on failure
print(f"  Name: {soul.name}\n")

print("Setting name to 'Souls'")  # Simulates regular use; ignore this chunk
soul.name = "Souls"  # only added as a sanity check to make sure
print(f"  Name: {soul.name}\n")  # the logic in this script is correct
print("Complete!")

```
### Console Output:
![image](https://user-images.githubusercontent.com/25145447/103483290-2b2a1e80-4e21-11eb-91e6-05c867b5c19d.png)

**Comment by KOOKIIEStudios** [comment]
## Suggested solution:
Add similar functionality to the setters, and provide documentation on how to propagate the error forward to the user in Discord,

Issue posted to log the issue.  
TODO for upcoming patches.

**Comment by KOOKIIEStudios** [comment]
Fixed as of commit b401b9ed4d677798f0331554370018f5648a04ff
